pkg/pulumiyaml/sort.go

```diff
@@ -73,6 +104,58 @@ func topologicallySortedResources(t *ast.TemplateDecl) ([]graphNode, syntax.Diag
 			sorted = append(sorted, node)
 		}
 	}
+	var projConf map[string]interface{}
```

Comment:
Instead of grabbing the config in `topologicallySortedResources`, could we adjust the signature to accept some additional config nodes.
```go
func topologicallySortedResources(t *ast.TemplateDecl, externalConfig []configNode) ([]graphNode, syntax.Diagnostics)
```

We should be able to handle all config nodes the same way, replacing `t.Config.Entries` with `append(extraConfig, templateConfig)` in the above `for` loop.

Here `templateConfig` is simply:
```go
templateConfig := make([]configNode, len(t.Configuration.Entries)
for i, kip := range t.Configuration.Entries {
    templateConfig[i] = configNode(kvp)
}
```

This gives us two benefits:
1. DRY: We always handle adding `configNode`s the same way.
2. `topologicallySortedResources` remains a pure function, making it much easier to test and maintain. We can put the `PULUMI_CONFIG -> []configNode` logic in a separate function.